### ISSUE TITLE
knative/pkg/ or knative.dev/pkg only supports version of kubernetes-1.12.x, not 1.13.x

### ISSUE THREAD

**Issue by houshengbo** [knative/serving-operator#60]
We have applied the following dependencies in operator:
```
[[override]]
  name = "k8s.io/code-generator"
  # revision for tag "kubernetes-1.13.1"
  revision = "c2090bec4d9b1fb25de3812f868accc2bc9ecbae"

[[override]]
  name = "k8s.io/api"
  # revision for tag "kubernetes-1.13.1"
  revision = "05914d821849570fba9eacfb29466f2d8d3cd229"

[[override]]
  name = "k8s.io/apiextensions-apiserver"
  # revision for tag "kubernetes-1.13.1"
  revision = "0fe22c71c47604641d9aa352c785b7912c200562"

[[override]]
  name = "k8s.io/apimachinery"
  # revision for tag "kubernetes-1.13.1"
  revision = "2b1284ed4c93a43499e781493253e2ac5959c4fd"

[[override]]
  name = "k8s.io/client-go"
  # revision for tag "kubernetes-1.13.1"
  revision = "8d9ed539ba3134352c586810e749e58df4e94e4f"


```
However, we will hit the issue along the line like 
https://github.com/knative/pkg/blob/master/client/clientset/versioned/typed/authentication/v1alpha1/fake/fake_policy.go#L122,
when we add the reconciler into this repo. 

Error logged when I run the test cases:
```
# github.com/knative/serving-operator/vendor/knative.dev/pkg/client/clientset/versioned/typed/authentication/v1alpha1/fake
vendor/knative.dev/pkg/client/clientset/versioned/typed/authentication/v1alpha1/fake/fake_policy.go:122:44: not enough arguments in call to testing.NewPatchSubresourceAction
	have (schema.GroupVersionResource, string, string, []byte, []string...)
	want (schema.GroupVersionResource, string, string, types.PatchType, []byte, ...string)
# github.com/knative/serving-operator/vendor/knative.dev/pkg/client/clientset/versioned/typed/istio/v1alpha3/fake
vendor/knative.dev/pkg/client/clientset/versioned/typed/istio/v1alpha3/fake/fake_destinationrule.go:122:44: not enough arguments in call to testing.NewPatchSubresourceAction
	have (schema.GroupVersionResource, string, string, []byte, []string...)
	want (schema.GroupVersionResource, string, string, types.PatchType, []byte, ...string)
vendor/knative.dev/pkg/client/clientset/versioned/typed/istio/v1alpha3/fake/fake_gateway.go:122:44: not enough arguments in call to testing.NewPatchSubresourceAction
	have (schema.GroupVersionResource, string, string, []byte, []string...)
	want (schema.GroupVersionResource, string, string, types.PatchType, []byte, ...string)
vendor/knative.dev/pkg/client/clientset/versioned/typed/istio/v1alpha3/fake/fake_virtualservice.go:122:44: not enough arguments in call to testing.NewPatchSubresourceAction
	have (schema.GroupVersionResource, string, string, []byte, []string...)
	want (schema.GroupVersionResource, string, string, types.PatchType, []byte, ...string)
```

One solution to this issue: manually edit the code under vendor/knative.dev/pkg to accommodate, as in https://github.com/kubernetes/cluster-registry/issues/271 or https://github.com/knative/client/issues/23.

**Comment by k4leung4** [comment]
Chatted with @mattmoor and while there is intent to upgrade knative/pkg to 1.13, it might take a bit of time as there are changes that needs time to resolve.
If you are willing to do the work, great, if not, we may have to take the patching vendor dir route.

**Comment by houshengbo** [comment]
@k4leung4 Thx.
There are something else I found regarding downgrading to 1.12.x.

There are currently two pakcgaes: "github.com/operator-framework/operator-sdk" and "github.com/jcrossley3/manifestival", hindering us from doing that.

We have to go back to v0.4.1 for github.com/operator-framework/operator-sdk to pin to kubernetes 1.12.x, which is a commit in Jan, 2019.

For "github.com/jcrossley3/manifestival", we do not have any commit supporting kubernetes 1.12.x.

**Comment by jcrossley3** [comment]
> We have to go back to v0.4.1 for github.com/operator-framework/operator-sdk to pin to kubernetes 1.12.x, which is a commit in Jan, 2019.

I see supporting `operator-sdk` while we transition to `knative.dev` as a nice-to-have luxury. I say rip the bandaid off!

> For "github.com/jcrossley3/manifestival", we do not have any commit supporting kubernetes 1.12.x.

That should be easy enough to fix. :smile:

**Comment by houshengbo** [comment]
In terms of the work effort, there is less hassle to patch knative.dev/pkg to accommodate 1.13.x than yo downgrade to 1.12.x.

**Comment by jcrossley3** [comment]
If it helps, I just created a `client-go` branch in the manifestival repo that replaces the controller-runtime client usage with client-go's dynamic lib. This allowed me to depend on the same version of k8s that knative-serving does: 1.12.9. I tried updating Gopkg.toml to refer to the `client-go` branch, but `dep ensure` failed with `Solving failure: package github.com/Azure/go-autorest/version does not exist within project github.com/Azure/go-autorest`.

Hopefully, you might have better luck: https://github.com/jcrossley3/manifestival/tree/client-go
